subsystem/exp_results/run.py

```diff
@@ -31,14 +32,30 @@ def failed_experiment_field(exp, stage_statuses, stage, duration=None, notify=No
 
     return build_field(title=exp, value=message)
 
+def send_graphs(config, info, client, output_dir):
+    os.chdir(output_dir)
+    for (curr_dir, _, files) in os.walk(info.exp_graphs):
+        for filename in files:
+            if filename.endswith('.png') or filename.endswith('.jpg'):
+                success, msg = upload_image(client, config['channel_id'], f'{curr_dir}/{filename}', filename)
```

Comment:
I think these uploaded images should have comment text (using [this method](https://api.slack.com/methods/files.upload#arg_initial_comment)) stating that they are automatic messages, as the API will upload them as a specific user, not as the bot.

I also think that we might be able to post the graphs more neatly in a thread using [the thread reply API](https://api.slack.com/methods/files.upload#arg_thread_ts). That way we can have a neater post using the following format:
- Top-level post giving the experiment name
- Post graph images as a reply
- (Maybe post the summary this way too? Then we can forego the giant, often-truncated message that lumps the summaries together)

Reply:
Great point! I will try to refactor the code in this way.